recipes/bx/all/conanfile.py

```diff
@@ -1,81 +1,189 @@
+from conan import ConanFile
+from conan.tools.files import copy, get, rename
+from conan.tools.build import check_min_cppstd
+from conan.tools.layout import basic_layout
+from conan.tools.microsoft import is_msvc
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.microsoft import MSBuild, VCVars
+from conan.tools.gnu import Autotools, AutotoolsToolchain
+from pathlib import Path
 import os
-import glob
-from conans import ConanFile, CMake, tools
-from conans.errors import ConanInvalidConfiguration
 
-class BxConan(ConanFile):
+required_conan_version = ">=1.50.0"
+
+
+class bxConan(ConanFile):
     name = "bx"
-    description = "Base library used across multiple projects."
     license = "BSD-2-Clause"
-    topics = ("conan", "bx")
     homepage = "https://github.com/bkaradzic/bx"
     url = "https://github.com/conan-io/conan-center-index"
-    exports_sources = ["CMakeLists.txt"]
-    generators = "cmake"
-    settings = "os", "arch", "compiler", "build_type"
-    options = {
-        "fPIC": [True, False],
-    }
-    default_options = {
-        "fPIC": True,
-    }
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
-    @property
-    def _build_subfolder(self):
-        return "build_subfolder"
-
-    def config_options(self):
-        if self.settings.os == 'Windows':
-            del self.options.fPIC
+    description = "Base library providing utility functions and macros."
+    topics = ("general", "utility")
+    settings = "os", "compiler", "arch", "build_type"
+    options = {"fPIC": [True, False], "tools": [True, False]}
+    default_options = {"fPIC": True, "tools": False}
+
+    bxFolder = "bx"
```

Comment:
```suggestion
    @property
    def _bx_folder(self):
        return "bx"
```
please, keep sneak_case to follow same pattern from other variables and methods. Also, any custom method or custom member attribute should be protected to avoid possible naming conflict with future conan versions.

Reply:
Makes sense, changed.